File: uva_729.py
def next_permutation(a):
    for i in reversed(range(len(a) - 1)): 
        if a[i] < a[i + 1]: 
            break
    else:
        return False
    j = next(j for j in reversed(range(i + 1, len(a))) if a[i] < a[j]) 
    a[i], a[j] = a[j], a[i] 
    a[i + 1:] = reversed(a[i + 1:]) 
    return True

num=int(input())
while num:
    num-=1
    temp=input()
    n, h=input().split()
    n, h=int(n), int(h)
    ans=[]
    for _ in range(n-h):
        ans.append('0')
    for _ in range(h):
        ans.append('1')
    while True:
        for item in ans:
            print(item, end='')
        print('')
        if not next_permutation(ans):
            break
    if num:
        print('')

File: uva_729.cpp
# include <iostream>
# include <algorithm>
using namespace std;
int main(void){
    int num, n, h;
    cin >> num;
    while(num--){
        char str[20];
        cin >> n >> h;
        for (int i=0; i<n-h; ++i)
            str[i]='0';
        for (int i=n-h; i<n; ++i)
            str[i]='1';
        str[n]='\0';
        do{
            cout << str << endl;
        }while(next_permutation(str, str + n));
        if (num)
            cout << endl;
    }
    return 0;
}